tensorflow/contrib/image/python/ops/image_ops.py

```diff
@@ -508,6 +511,82 @@ def has_zero():
     else:
       return components
 
+def recursive_fill(mask, threshold, total_dims, max_iter=None):
+  """Recursively iterates until the provided 0-1s 2D or 3D mask is filled.
+  If max_iter is provided, it will iterate for a number of times.
+  If total_dims is 2, threshold is recommended to be 4.
+  If total_dims is 3, threshold is recommended to be 13.
+
+  Args:
+    mask: A 2D or 3D `Tensor` of 0s and 1s representing the mask to be filled.
+    threshold: A threshold indicating sensitive the hole detector will be.
+    total_dims: Number of dimensions of the mask. Only 2 or 3 allowed.
```

Comment:
We would normally get the static shape of the mask and require that the number of dimensions is set, and not pass the dimensionality as a separate argument. For example, see around line 250 of this file. Then if the number of dimensions is statically unknown, the caller can manually set it with e.g. `mask.set_shape([None, None])` for 2D.